src/types/redis_hash.cc

```diff
@@ -418,4 +472,208 @@ rocksdb::Status Hash::RandField(const Slice &user_key, int64_t command_count, st
   return rocksdb::Status::OK();
 }
 
+rocksdb::Status Hash::ExpireFields(const Slice &user_key, uint64_t expire_ms, const std::vector<Slice> &fields,
+                                   HashFieldExpireType type, bool is_persist, std::vector<int8_t> *ret) {
+  std::string ns_key = AppendNamespacePrefix(user_key);
+  HashMetadata metadata(false);
+  LatestSnapShot ss(storage_);
+  rocksdb::Status s = GetMetadata(GetOptions{ss.GetSnapShot()}, ns_key, &metadata);
+  if (!s.ok()) {
+    ret->resize(fields.size(), -2);
+    return rocksdb::Status::OK();
+  }
+
+  rocksdb::ReadOptions read_options = storage_->DefaultMultiGetOptions();
+  read_options.snapshot = ss.GetSnapShot();
+
+  std::vector<rocksdb::Slice> keys;
+  keys.reserve(fields.size());
+  std::vector<std::string> sub_keys;
+  sub_keys.resize(fields.size());
+  for (size_t i = 0; i < fields.size(); i++) {
+    auto &field = fields[i];
+    sub_keys[i] = InternalKey(ns_key, field, metadata.version, storage_->IsSlotIdEncoded()).Encode();
+    keys.emplace_back(sub_keys[i]);
+  }
+
+  auto batch = storage_->GetWriteBatchBase();
+  WriteBatchLogData log_data(kRedisHash);
+  batch->PutLogData(log_data.Encode());
+
+  // expire special field
+  std::vector<rocksdb::PinnableSlice> values_vector;
+  values_vector.resize(sub_keys.size());
+  std::vector<rocksdb::Status> statuses_vector;
+  statuses_vector.resize(sub_keys.size());
+  storage_->MultiGet(read_options, storage_->GetDB()->DefaultColumnFamily(), keys.size(), keys.data(),
+                     values_vector.data(), statuses_vector.data());
+
+  auto now = util::GetTimeStampMS();
+  for (size_t i = 0; i < keys.size(); i++) {
+    if (!statuses_vector[i].ok() && !statuses_vector[i].IsNotFound()) return statuses_vector[i];
+
+    // no such field exists
+    if (statuses_vector[i].IsNotFound()) {
+      ret->emplace_back(-2);
+      continue;
+    }
+
+    InternalKey sub_ikey(ns_key, fields[i], metadata.version, storage_->IsSlotIdEncoded());
+
+    // expire with a pass time
+    if (expire_ms <= now && !is_persist) {
+      batch->Delete(sub_ikey.Encode());
+      ret->emplace_back(2);
```

Comment:
Should be `-2`？

Reply:
According to https://redis.io/docs/latest/commands/hexpire/ 

>Array reply. For each field:
Integer reply: -2 if no such field exists in the provided hash key, or the provided key does not exist.
Integer reply: 0 if the specified NX | XX | GT | LT condition has not been met.
Integer reply: 1 if the expiration time was set/updated.
Integer reply: 2 when HEXPIRE/HPEXPIRE is called with 0 seconds/milliseconds or when HEXPIREAT/HPEXPIREAT is called with a past Unix time in seconds/milliseconds.

expire with a pass time should return `2`? please let me know if I'm wrong.

Reply:
Got it, thanks.